packages/flagging/src/openfeature/provider.ts

```diff
@@ -5,57 +5,68 @@
 import type { EvaluationContext, JsonValue, Logger, Paradigm, ResolutionDetails } from '@openfeature/core'
 import { StandardResolutionReasons } from '@openfeature/core'
 import type { Provider } from '@openfeature/web-sdk'
+import type { PrecomputeClient } from '../precomputeClient'
 
 export class DatadogProvider implements Provider {
   runsOn: Paradigm = 'client'
   metadata = {
     name: 'datadog',
   } as const
 
+  private precomputeClient: PrecomputeClient | undefined
+
+  constructor(precomputeClient?: PrecomputeClient) {
+    this.precomputeClient = precomputeClient
+  }
+
```

Comment:
The API here is still rough and needs refinement as we add network backed precompute loading mode.